v2/svid/x509svid/svid.go

```diff
@@ -25,38 +32,145 @@ type SVID struct {
 // Load loads the X509-SVID from PEM encoded files on disk. certFile and
 // keyFile may be the same file.
 func Load(certFile, keyFile string) (*SVID, error) {
-	panic("not implemented")
+	certBytes, err := ioutil.ReadFile(certFile)
+	if err != nil {
+		return nil, x509SVIDErr.New("cannot read certificate file: %w", err)
+	}
+
+	keyBytes, err := ioutil.ReadFile(keyFile)
+	if err != nil {
+		return nil, x509SVIDErr.New("cannot read key file: %w", err)
+	}
+
+	return Parse(certBytes, keyBytes)
 }
 
 // Parse parses the X509-SVID from PEM blocks containing certificate and key
 // bytes. The certificate must be one or more PEM blocks with ASN.1 DER. The
 // key must be a PEM block with PKCS#8 ASN.1 DER.
 func Parse(certBytes, keyBytes []byte) (*SVID, error) {
-	panic("not implemented")
+	certs := []*x509.Certificate{}
+	for {
+		if len(certBytes) == 0 {
+			break
+		}
+		cert, rest, err := pemutil.ParseCertificate(certBytes)
+		certBytes = rest
+		if errors.Is(err, pemutil.ErrUnexpectedBlockType) {
+			continue
+		}
+		if err != nil {
+			return nil, x509SVIDErr.New("cannot parse PEM encoded certificate: %v", err)
+		}
+		certs = append(certs, cert)
+	}
+
+	var privateKey crypto.PrivateKey
+	var err error
+	for {
+		if len(keyBytes) == 0 {
+			break
+		}
+		privateKey, keyBytes, err = pemutil.ParsePrivateKey(keyBytes)
+		if errors.Is(err, pemutil.ErrUnexpectedBlockType) {
+			continue
+		}
+		if err != nil {
+			return nil, x509SVIDErr.New("cannot parse PEM encoded private key: %v", err)
+		}
+		break
+	}
+
+	return newSVID(certs, privateKey)
 }
 
 // ParseRaw parses the X509-SVID from certificate and key bytes. The
 // certificate must be ASN.1 DER (concatenated with no intermediate
 // padding if there are more than one certificate). The key must be a PKCS#8
 // ASN.1 DER.
 func ParseRaw(certBytes, keyBytes []byte) (*SVID, error) {
-	panic("not implemented")
+	certificates, err := x509.ParseCertificates(certBytes)
+	if err != nil {
+		return nil, x509SVIDErr.New("cannot parse DER encoded certificate: %v", err)
+	}
+
+	privateKey, err := x509.ParsePKCS8PrivateKey(keyBytes)
+	if err != nil {
+		return nil, x509SVIDErr.New("cannot parse DER encoded private key: %v", err)
+	}
+
+	return newSVID(certificates, privateKey)
 }
 
 // Marshal marshals the X509-SVID and returns PEM encoded blocks for the SVID
 // and private key.
 func (s *SVID) Marshal() ([]byte, []byte, error) {
-	panic("not implemented")
+	certBytes := pemutil.EncodeCertificates(s.Certificates)
```

Comment:
I was wondering... should fail to marshal an empty slice of certificates? 
(Same question for `MarshalRaw` method)

Reply:
Hmm. That's a good question. On one hand, the SVID will be validated already when it is loaded or sourced from the Workload API. On the other hand, it seems like an OK defensive measure. It would be easy to relax it later if we wanted. I'd say go ahead and do that.